web/server/entrypoint.go

```diff
@@ -328,6 +329,7 @@ func (s *robotServer) serveWeb(ctx context.Context, cfg *config.Config) (err err
 				restartInterval = newRestartInterval
 
 				if mustRestart {
+					s.logger.Debugf("backtrace at robot restart, %s", string(debug.Stack()))
```

Comment:
Ah, this is just printing out the stack trace of one goroutine. You [should use `runtime.Stack(buf, true)`](https://pkg.go.dev/runtime#Stack) to get the string representing all stack traces.

Obligatory [stackoverflow example](https://stackoverflow.com/a/27398062). Note initializing a large buffer. The while-true loop is unnecessary as this is intended to only be called once.

 Maybe warn if the return value of `runtime.Stack` (number of bytes written) equals the buffer size. Meaning the results are truncated. Simply printing "Warning: stack trace is truncated to %v bytes" followed by the stack traces is fine by me.

Reply:
Ooh thanks for this! will update :)